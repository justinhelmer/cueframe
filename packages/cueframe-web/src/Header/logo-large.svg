<svg
  xmlns="http://www.w3.org/2000/svg"
  viewBox="0 0 206.1 24.8">
  <defs>
    <style>.cls-1{fill:#ff5b1a;}.cls-2{fill:#fff;}.cls-3{fill:none;}</style>
  </defs>
  <title>logo</title>
  <g id="blocks">
    <path class="cls-1" d="M182.94,25.12h-26v-24h26Z" transform="translate(-2.24 -0.8)"/>
    <rect class="cls-2" x="157.13" y="2.33" width="3" height="20" rx="1.5"/>
    <rect class="cls-2" x="175.27" y="2.33" width="3" height="20" rx="1.5"/>
    <rect class="cls-2" x="166.2" y="9.03" width="3" height="13.3" rx="1.5"/>
    <rect class="cls-2" x="161.66" y="5.43" width="3" height="13.3" rx="1.5"/>
    <rect class="cls-2" x="170.73" y="5.43" width="3" height="13.3" rx="1.5"/>
  </g>
  <g id="vertical_lines" data-name="vertical lines">
    <path class="cls-3" d="M-77.95,81.89" transform="translate(-2.24 -0.8)"/>
  </g>
  <g id="text">
    <path
      class="cls-1"
      d="M2.24,11.53A15.08,15.08,0,0,1,3,6.43,8.4,8.4,0,0,1,5.08,3.12,7.82,7.82,0,0,1,8.25,1.34a14.23,14.23,0,0,1,4-.54,21.61,21.61,0,0,1,4.5.47V6A11.3,11.3,0,0,0,15,5.57a17.3,17.3,0,0,0-1.84-.09,9.28,9.28,0,0,0-1.78.16,3,3,0,0,0-1.48.74A4,4,0,0,0,8.84,8a9,9,00,0,0-.37,2.9v4.35a8.66,8.66,0,0,0,.39,2.9,4,4,0,0,0,1.05,1.64,3.14,3.14,0,0,0,1.49.74,10.16,10.16,0,0,0,1.78.16,18.76,18.76,0,0,0,2.07-.11,12.63,12.63,0,0,0,1.68-.29v4.61a18.44,18.44,0,0,1-2.29.36c-.76.07-1.52.11-2.29.11a14.48,14.48,0,0,1-4-.54,8.05,8.05,0,0,1-3.2-1.78A8.36,8.36,0,0,1,3,19.75a15,15,0,0,1-.75-5.09Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M35.83,25.6A12.83,12.83,0,0,1,32,25.08a6.43,6.43,0,0,1-2.77-1.71,7.67,7.67,0,0,1-1.69-3.11,16.9,16.9,0,0,1-.58-4.74V1.09h6V15c0,1.1,0,2,.09,2.81a6.19,6.19,0,0,0,.4,1.89,2.11,2.11,0,0,0,.86,1.06,3.65,3.65,0,0,0,3,0,2.11,2.11,0,0,0,.86-1.06,6.19,6.19,0,0,0,.4-1.89c.06-.77.09-1.71.09-2.81V1.09h6V15.42a18.53,18.53,0,0,1-.54,4.86,7.52,7.52,0,0,1-1.64,3.13,6.14,6.14,0,0,1-2.77,1.69A13.62,13.62,0,0,1,35.83,25.6Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M55.66,1.09h14V5.7H61.42v4.86h7.45v4.6H61.42v5.33h8.5V25.1H55.66Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M79.93,1.09H93.82V5.7H85.76V11h7.31v4.68H85.76V25.1H79.93Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M103.83,1.09h8.71q7.27,0,7.27,7.16a8.64,8.64,0,0,1-.59,3.37,4.87,4.87,0,0,1-2,2.28,3.75,3.75,0,0,1,1.47,1.84,7.2,7.2,0,0,1,.44,2.56v6.8h-5.69V18.69a4.68,4.68,0,0,0-.31-1.73,1.36,1.36,0,0,0-1.42-.75h-2.23V25.1h-5.69Zm8.1,10.83a1.6,1.6,0,0,0,1.37-.68,4.3,4.3,0,0,0,.5-2.41,4.33,4.33,0,0,0-.52-2.52,1.74,1.74,0,0,0-1.46-.69h-2.3v6.3Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M134.9,1.09h6.33l6.3,24h-5.72l-1.15-5.29H135l-1.19,5.29h-5.51Zm4.71,14.29-1.76-8.06-1.8,8.06Z"
      transform="translate(-2.24 -0.8)"
    />
    <path
      class="cls-1"
      d="M194.08,1.09h14V5.7h-8.21v4.86h7.45v4.6h-7.45v5.33h8.5V25.1H194.08Z"
      transform="translate(-2.24 -0.8)"
    />
  </g>
</svg>