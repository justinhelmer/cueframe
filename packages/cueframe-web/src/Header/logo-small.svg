<svg
  xmlns="http://www.w3.org/2000/svg"
  viewBox="0 0 26 24.8">
  <defs>
    <style>.cls-1{fill:#ff5b1a;}.cls-2{fill:#fff;}</style>
  </defs>
  <title>logo</title>
  <g id="blocks">
    <rect class="cls-1" x="0" y="0" width="100%" height="100%" />
    <rect class="cls-2" x="2.5" y="2.33" width="3" height="20" rx="1.5"/>
    <rect class="cls-2" x="7" y="5.43" width="3" height="13.3" rx="1.5"/>
    <rect class="cls-2" x="11.5" y="9.03" width="3" height="13.3" rx="1.5"/>
    <rect class="cls-2" x="16" y="5.43" width="3" height="13.3" rx="1.5"/>
    <rect class="cls-2" x="20.5" y="2.33" width="3" height="20" rx="1.5"/>
  </g>
</svg>