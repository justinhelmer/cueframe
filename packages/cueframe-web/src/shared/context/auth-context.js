import React, { useState, useLayoutEffect } from 'react';
import { withCookies } from 'react-cookie';

const AuthContext = React.createContext();

const AuthProvider = withCookies(({ children, cookies }) => {
  const uidCookie = cookies.get('user.id');
  const [isAuthenticated, setIsAuthenticated] = useState(uidCookie);
  const [user, setUser] = useState({});
  const [userDataReady, setUserDataReady] = useState();

  const setUserSession = async (resp) => {
    setUser(await resp.json());
    setIsAuthenticated(true);
  };

  const killUserSession = () => {
    cookies.remove('user.id');
    setUser({});
    setIsAuthenticated(false);
  }

  const login = async (email, password) => {
    const resp = await fetch('/api/user/login', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ email, password }),
    });

    if (resp.ok) {
      await setUserSession(resp);
    } else {
      const err = new Error('User login failed');
      err.resp = resp;
      throw err;
    }
  };

  const logout = async () => {
    const resp = await fetch('/api/user/logout', { method: 'POST' });

    if (resp.ok) {
      killUserSession();
    } else {
      const err = new Error('User logout failed');
      err.resp = resp;
      throw err;
    }
  };

  // on a page refresh, if the user has a logged in cookie, fetch user data. use layout effect so the fetch promise can be passed to context consumers
  useLayoutEffect(() => {
    if (uidCookie && !user.id) {
      const promise = fetch(`/api/user/${uidCookie}`)
        .then((resp) => {
          if (resp.ok) {
            return setUserSession(resp);
          } else {
            killUserSession();
          }
        });

      setUserDataReady(promise);
    } else {
      setUserDataReady(Promise.resolve());
    }
  }, []);

  return (
    <AuthContext.Provider value={{
      isAuthenticated,
      userDataReady,
      user,
      login,
      logout,
    }}>
      {children}
    </AuthContext.Provider>
  );
});

export { AuthContext, AuthProvider };