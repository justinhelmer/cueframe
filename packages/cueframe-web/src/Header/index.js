import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import { MDBNavbar, MDBNavbarBrand, MDBNavbarNav, MDBNavItem, MDBNavLink } from 'mdbreact';
import { AuthContext } from '../shared/context/auth-context';
import logoSmall from './logo-small.svg';
import logoLarge from './logo-large.svg';
import './header.scss';

export default () => {
  const { isAuthenticated, logout } = useContext(AuthContext);

  return (
    <MDBNavbar id="header" dark color="primary-color">
      <MDBNavbarBrand>
        <Link className="text-hide d-block" to="/">
          <img src={logoSmall} alt="Cueframe" className="d-block d-md-none" />
          <img src={logoLarge} alt="Cueframe" className="d-none d-md-block" />
          Cueframe
        </Link>
      </MDBNavbarBrand>
      <MDBNavbarNav right>
        <MDBNavItem>
          {isAuthenticated ? (
            <MDBNavLink to="#" onClick={logout}>Logout</MDBNavLink>
          ) : (
            <MDBNavLink to="/login">Login</MDBNavLink>
          )}
        </MDBNavItem>
      </MDBNavbarNav>
    </MDBNavbar>
  );
};
