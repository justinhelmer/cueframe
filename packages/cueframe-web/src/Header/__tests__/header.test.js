import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { AuthContext } from '../../shared/context/auth-context';
import Header from '..';

describe('<Header/>', () => {
  describe('Snapshot', () => {
    it('renders correctly for logged out users', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <AuthContext.Provider value={{}}>
              <Header/>
            </AuthContext.Provider>
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it('renders correctly for logged in users', () => {
      const tree = renderer
        .create(
          <MemoryRouter>
            <AuthContext.Provider value={{ isAuthenticated: true }}>
              <Header/>
            </AuthContext.Provider>
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  describe('Component', () => {
    it('fires logout methos from auth context when logout is clicked', () => {
      const logout = jest.fn();
      const wrapper = mount(
        <MemoryRouter>
          <AuthContext.Provider value={{ isAuthenticated: true, logout }}>
            <Header/>
          </AuthContext.Provider>
        </MemoryRouter>
      );

      expect(logout).not.toHaveBeenCalled();
      wrapper.findWhere(n => n.type() === 'a' && n.text() === 'Logout').simulate('click');
      expect(logout).toHaveBeenCalled();
    })
  });
});