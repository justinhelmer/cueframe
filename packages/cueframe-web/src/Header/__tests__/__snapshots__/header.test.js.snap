// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Snapshot renders correctly for logged in users 1`] = `
<nav
  className="navbar navbar-dark primary-color"
  id="header"
  role="navigation"
>
  <div
    className="navbar-brand"
  >
    <a
      className="text-hide d-block"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Cueframe"
        className="d-block d-md-none"
        src="logo-small.svg"
      />
      <img
        alt="Cueframe"
        className="d-none d-md-block"
        src="logo-large.svg"
      />
      Cueframe
    </a>
  </div>
  <ul
    className="navbar-nav ml-auto"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link Ripple-parent"
        href="/"
        onClick={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        Logout
        <div
          className="Ripple "
          style={
            Object {
              "height": "0px",
              "left": "0px",
              "top": "0px",
              "width": "0px",
            }
          }
        />
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Header/> Snapshot renders correctly for logged out users 1`] = `
<nav
  className="navbar navbar-dark primary-color"
  id="header"
  role="navigation"
>
  <div
    className="navbar-brand"
  >
    <a
      className="text-hide d-block"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Cueframe"
        className="d-block d-md-none"
        src="logo-small.svg"
      />
      <img
        alt="Cueframe"
        className="d-none d-md-block"
        src="logo-large.svg"
      />
      Cueframe
    </a>
  </div>
  <ul
    className="navbar-nav ml-auto"
  >
    <li
      className="nav-item"
    >
      <a
        aria-current={null}
        className="nav-link Ripple-parent"
        href="/login"
        onClick={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        Login
        <div
          className="Ripple "
          style={
            Object {
              "height": "0px",
              "left": "0px",
              "top": "0px",
              "width": "0px",
            }
          }
        />
      </a>
    </li>
  </ul>
</nav>
`;
